sled-agent/src/hardware/illumos/mod.rs

```diff
@@ -73,56 +170,187 @@ fn node_name(subsystem_vid: i32, subsystem_id: i32) -> String {
     format!("pci{subsystem_vid:x},{subsystem_id:x}")
 }
 
+fn slot_to_disk_variant(slot: i64) -> Option<DiskVariant> {
+    match slot {
+        0x00..=0x09 => Some(DiskVariant::U2),
+        0x11..=0x12 => Some(DiskVariant::M2),
```

Comment:
Where do these values come from?

Reply:
https://github.com/oxidecomputer/illumos-gate/blob/stlouis/usr/src/uts/oxide/milan/milan_dxio_data.c#L823-L847

Added a link to this in the documentation for this function.